src/extensions/reactNative.ts

```diff
@@ -1,36 +1,27 @@
-const { test, trim } = require('ramda')
-const exitCodes = require('../lib/exitCodes')
+import { test, trim } from 'ramda'
+import exitCodes from '../lib/exit-codes'
+import { IgniteToolbox, IgniteRNInstallResult } from '../types'
 
 // DEPRECATED: Please specify React Native version when invoking install
 // Example: const rnInstall = await reactNative.install({ name, version: '0.42.0' })
-const REACT_NATIVE_VERSION = '0.42.0'
+const REACT_NATIVE_VERSION = '0.58.0'
 
 /**
- * Attach this extension to the context.
- *
- * @param   {any} plugin  - The current running plugin.
- * @param   {any} command - The current running command.
- * @param   {any} context - The active context.
- * @returns {any}         - The public-facing features this supports.
+ * Attach this extension to the toolbox.
  */
-function attach (plugin, command, context) {
+function attach(toolbox: IgniteToolbox) {
   // fist-full o features
-  const { parameters, print, system, filesystem, strings, ignite } = context
+  const { parameters, print, system, filesystem, strings, ignite } = toolbox
   const { log } = ignite
 
   /**
    * Installs React Native.
-   *
-   * @param {Object}  opts          - The options to pass to install react native.
-   * @param {string}  opts.name     - What to call this project.
-   * @param {string}  opts.version  - The React Native version to install (if we care).
-   * @param {string}  opts.template - The template to pass to React Native (if any).
-   * @param {boolean} opts.skipJest - Should we bypass jest?
-   * @returns {number}              - The error code we should exit with if non-0.
    */
-  async function install (opts = {}) {
+  async function install(
+    opts: { name?: string; version?: string; template?: string; skipJest?: boolean; useNpm?: boolean } = {},
+  ): Promise<IgniteRNInstallResult> {
     //  grab the name & version
-    const name = opts.name || parameters.third
+    const name = opts.name || parameters.first
```

Comment:
Gluegun prior to 2.0 had a bug where there were two spurious parameters, thus the `third`.

Reply:
Probably answers my questions above too.